bin/NativeTests/FunctionExecutionTest.cpp

```diff
@@ -0,0 +1,313 @@
+        //0d chakra!Js::InterpreterStackFrame::ProcessUnprofiled
+        //0e chakra!Js::InterpreterStackFrame::Process
+        //0f chakra!Js::InterpreterStackFrame::InterpreterHelper
+        //10 chakra!Js::InterpreterStackFrame::InterpreterThunk
+        //11 0x0
+        //12 chakra!amd64_CallFunction
+        //13 chakra!Js::JavascriptFunction::CallFunction<1>
+        //14 chakra!Js::JavascriptFunction::CallRootFunctionInternal
+        //15 chakra!Js::JavascriptFunction::CallRootFunction
+        //16 chakra!ScriptSite::CallRootFunction
+        //17 chakra!ScriptSite::Execute
+        //18 chakra!ScriptEngine::ExecutePendingScripts
+        //19 chakra!ScriptEngine::ParseScriptTextCore
+        //1a chakra!ScriptEngine::ParseScriptText
+        f.TryTransitionToJitExecutionMode();
+        CHECK(f.GetExecutionMode() == ExecutionMode::SimpleJit);
+        CHECK(f.GetInterpretedCount() == 0);
+
+        // Simple JIT EntryPoint Info keeps a count from the limit and decrements per call.
+        // Since the stub entry point is initialized to 0, proceed with transition.
+
+        //00 chakra!Js::FunctionExecutionStateMachine::SetExecutionMode
+        //01 chakra!Js::FunctionExecutionStateMachine::TryTransitionToNextExecutionMode
+        //02 chakra!Js::FunctionBody::TryTransitionToNextExecutionMode
+        //03 chakra!NativeCodeGenerator::TransitionFromSimpleJit
+        //04 chakra!NativeCodeGenerator::Jit_TransitionFromSimpleJit
+        //05 0x0
+        //06 chakra!amd64_CallFunction
+        //07 chakra!Js::JavascriptFunction::CallFunction<1>
+        //08 chakra!Js::InterpreterStackFrame::OP_CallCommon<Js::OpLayoutDynamicProfile<Js::OpLayoutT_CallIWithICIndex<Js::LayoutSizePolicy<0> > > >
+        //09 chakra!Js::InterpreterStackFrame::OP_CallI<Js::OpLayoutDynamicProfile<Js::OpLayoutT_CallIWithICIndex<Js::LayoutSizePolicy<0> > > >
+        //0a chakra!Js::InterpreterStackFrame::ProcessUnprofiled
+        //0b chakra!Js::InterpreterStackFrame::Process
+        //0c chakra!Js::InterpreterStackFrame::InterpreterHelper
+        //0d chakra!Js::InterpreterStackFrame::InterpreterThunk
+        //0e 0x0
+        //0f chakra!amd64_CallFunction
+        //10 chakra!Js::JavascriptFunction::CallFunction<1>
+        //11 chakra!Js::JavascriptFunction::CallRootFunctionInternal
+        //12 chakra!Js::JavascriptFunction::CallRootFunction
+        //13 chakra!ScriptSite::CallRootFunction
+        //14 chakra!ScriptSite::Execute
+        //15 chakra!ScriptEngine::ExecutePendingScripts
+        //16 chakra!ScriptEngine::ParseScriptTextCore
+        //17 chakra!ScriptEngine::ParseScriptText
+        f.TryTransitionToNextExecutionMode();
+        CHECK(f.GetExecutionMode() == ExecutionMode::FullJit);
+        CHECK(f.GetInterpretedCount() == 0);
+    }
+
+    // test what happens when we jit a Loop Body
+    TEST_CASE("FuncExe_NormalExecutionOfLoop")
```

Comment:
is this work in progress?

Reply:
yes--i'll followup with this alongside product code